src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/DefaultRazorDynamicFileInfoProvider.cs

```diff
@@ -140,14 +185,14 @@ public Task RemoveDynamicFileInfoAsync(ProjectId projectId, string projectFilePa
 
         private DynamicFileInfo CreateEmptyInfo(Key key)
         {
-            var filename = Path.ChangeExtension(key.FilePath, ".g.cs");
+            var filename = key.FilePath + ".g.cs";
```

Comment:
This seems like a change in our assumption. Does the existing non-LSP stuff call into this? If so, does that still work?

Reply:
The existing LSP stuff does but all of our infrastructure pieces rely on the `.g.cs` ending, nothing before it.